extensions/legit/porcelain-fossil.lisp

```diff
@@ -0,0 +1,137 @@
+(uiop:define-package :lem/porcelain-fossil
+  (:use :cl :lem/porcelain)
+  (:shadow :push)
+  (:import-from :trivial-types
+                :proper-list)
+  (:import-from :lem/porcelain
+                :porcelain-error)
+  (:export :fossil-project-p)
+  (:documentation "Implements the porcelain interface for fossil-based repos."))
+
+(in-package :lem/porcelain-fossil)
+
+(declaim (type (proper-list string) *fossil-base-args*))
+(defvar *fossil-base-args* (list "fossil")
+  "The fossil program, to be appended command-line options.")
+
+;; VCS implementation for fossil
+(defclass vcs-fossil () ())
```

Comment:
This calls for a parent dummy `vcs` class, doesn't it?

Reply:
Maybe, could be a home for that `project-root` member you talk about elsewhere. But (IIRC) CLOS is slightly weird in that you can just define methods for *any* generic function on *any* parameters, so it's not strictly necessary.

Reply:
I think a parent "vcs" class is important (for clarity, to add common slots, etc).

Also naming is important, `(defclass vcs-project () ())` might be more meaningful than just `vcs`.